### DISCUSSION TITLE
bearer != Bearer

### DISCUSSION THREAD

**Discussion by adamcstephens** [kanidm/kanidm#2344] [Q&A] [answered]
I'm sure you can guess what this is about, given the title, but I wanted to raise this discussion again. I have read https://github.com/kanidm/kanidm/discussions/2322 and understand what was said there, but unfortunately I'm having the same problem. This time it's with an app using the library https://github.com/panva/node-openid-client . Looking through their issues I discovered that they may not be interested in a solution that rewrites the token type as was done for JupyterHub, or at least they [weren't a few years ago](https://github.com/panva/node-openid-client/issues/248#issuecomment-611378183).

So how can we proceed? Is the only option updating downstream apps and libraries?

Is there an alternative of sending `Bearer` in the response, so that apps/libs that re-use token type can send that back and everything work?

I don't wish to undermine following of standards, and do think that we should follow them, but I also want to use kanidm with my applications. I'll understand if this is again a "won't fix", but as just a user of these tools it's sad that we're separated by the casing of a single letter.

**Comment by Firstyear** [comment]
I think the problem is if we introduce this work around, we also risk breaking other things that expect it to be uppercased. Let me do some more research and checking

**Reply by yaleman** [reply]
We can only try I guess, and if it breaks more than it fixes then ... fun!

**Reply by Firstyear** [reply]
If anything else breaks, I will literally cry.

**Reply by yaleman** [reply, marked as answer]
the RFC's are annoyingly vague 😞
